viewing_party/party.py

```diff
@@ -0,0 +1,235 @@
+        count_movies = 0
+        sum_of_ratings = 0
+        for movie in movies:
+            sum_of_ratings += movie["rating"]
+            count_movies += 1
+        
+        avg_rating = sum_of_ratings/count_movies
+        return avg_rating
+
+def get_most_watched_genre(user_data):
+    #starting with the list of movies
+    movies = user_data["watched"]
+
+    if len(movies) == 0:
+        return None
+    else:
+        list_genres = []
+        most_watched = ""
+        
+        for movie in movies:
+            list_genres.append(movie["genre"])
+
+        most_watched = Counter(list_genres).most_common(1)[0][0]
+
+    
+    return most_watched
+
+### WAVE 3 ###
+def get_unique_watched(user_data):
+    #first let's get the list of movies the user has watched
+    users_movies = []
+    for index in range(len(user_data["watched"])):    
+        users_movies.append(user_data["watched"][index]["title"])
+    
+    #print(users_movies)
+    
+
+    #print(user_data["friends"][0]["watched"][0]["title"])
+
+    friends_movies = set() #looping to fill this into a set
+
+    friends = user_data["friends"]
+    for friend in range(len(friends)):
+        watched_list = friends[friend]["watched"]
+        for movies in watched_list:
+            for individual_title in movies:
+                #print(movies[individual_title])
+                friends_movies.add(movies[individual_title])
+                
+    #print(friends_movies)
+
+    users_movies = set(users_movies)
+    
+    movies_only_user_watched = users_movies - friends_movies
+    #print(movies_only_user_watched)
+    
+    final_movie_list_of_dictionaries = []
+    for movie in movies_only_user_watched:
+        final_movie_list_of_dictionaries.append({"title": movie}) 
+    
+    return final_movie_list_of_dictionaries
+
+def get_friends_unique_watched(user_data):
+    #first, let's get the list of movies the user has watched
+    users_movies = []
+    for index in range(len(user_data["watched"])):    
+        users_movies.append(user_data["watched"][index]["title"])
+
+    #next, let's get the set of movies the friends have watched
+    friends_movies = set()
+
+    friends = user_data["friends"]
+    for friend in range(len(friends)):
+        watched_list = friends[friend]["watched"]
+        for movies in watched_list:
+            for individual_title in movies:
+                #print(movies[individual_title])
+                friends_movies.add(movies[individual_title])
+                
+    #convert the users_movies to a set for comparison
+    users_movies = set(users_movies)
+    
+    #find the difference
+    movies_at_least_one_friend_watched = friends_movies - users_movies
+
+    #convert to a list of dictionaries to return
+    final_movie_list_of_dictionaries = []
+    for movie in movies_at_least_one_friend_watched:
+        final_movie_list_of_dictionaries.append({"title": movie}) 
+    
+    return final_movie_list_of_dictionaries
```

Comment:
This function is structurally the same as get_unique_watched.  I recommend thinking about how you could create a helper function that takes two lists and does this work.  You could then call the helper function from `get_unique_watched` and `get_friends_unique_watched`.